minbpe/regex.py

```diff
@@ -41,17 +41,26 @@ def train(self, text, vocab_size, verbose=False):
         text_chunks = re.findall(self.compiled_pattern, text)
```

Comment:
Faster to do the counting ahead of converting to unicode
```suggestion
        text_chunks = re.findall(self.compiled_pattern, text)
        chunks_counted = collections.Counter(text_chunks)
        text_chunks = [chunk for chunk, count in chunks_counted.items()]
        global_counts = [count for chunk, count in chunks_counted.items()]
```

Then further down we can just go:
```
            for chunk_ids, global_count in zip(ids, global_counts):
                # passing in stats will update it in place, adding up counts
                get_stats_n(chunk_ids, stats, global_count)
```